confirm in question now accepts Proc

mmihira opened the pull request:
Implemented functionality to assign a Proc object to the confirm attribute of the Highline::Question object.

abinoam commented:
Closes #201 

See:
- https://github.com/blog/1506-closing-issues-via-pull-requests
- https://help.github.com/articles/closing-issues-via-commit-messages/

abinoam commented:
Could you add a test for it?

Could you also please update the documentation (comment) adding this new functionality?
https://github.com/JEG2/highline/blob/master/lib/highline/question.rb#L149

And also update the Changelog.md so it cites the addition and gives credit to you?

If you need any assistance in any of these, please tell me, I would be glad to help you along.

And thank you for the pull request, we'll be merging it as soon as it's complete. 👍

mmihira commented:
I just made a test, updated the docs and also added an entry in the change log.
Thank you for the clear guidance. Let me know if there are any other issues.

abinoam commented on lib/highline/question.rb:
```diff
@@ -147,11 +147,15 @@ def initialize(template, answer_type)
+    #               output a string which is then used as the confirm 
+    #               question. 
+    # String  :     The String must use ERB syntax. The String is 
+    #               evaluated with with access to question and answer and
```
Hi @mmihira, a small typo here. Double _with_. In _evaluated with with access_.

abinoam commented:
Great @mmihira !

abinoam commented:
As soon as you fix the tiny typo we're gonna merge. It'll be perfect with tests and everything. Thank you in advance.

mmihira commented:
Thanks @abinoam. I fixed up the typo.

abinoam commented:
I have merged it into master. It will be on the next release.
Thank you very much @mmihira! 👍
